### ISSUE TITLE
Dynamic column width

### ISSUE THREAD

**Issue by arthur-jacquin** [numworks/epsilon#1738]
#### Problem you'd like to fix

There are several Table/Data/Stats tabs across Epsilon. Unfortunately, the column width is static. Depending on the content of the column, there might be quite a lot a space wasted. A better view could help "make the user a math person".

In the following example, the second regression is not visible while the X1 and X2 columns only contains small-sized integers:

![column_width](https://user-images.githubusercontent.com/53219173/102717380-47d33c00-42e2-11eb-95fc-01b8144aed1d.png)

#### Describe the solution you'd like
Dynamic column width.

**Comment by Ecco** [comment]
This is a solid suggestion, thanks @arthur-jacquin !

I agree that the width of a column should be the same as its widest element. But then this means we would need to iterate over every single possible value of each column, compute its width, and then set the column's value.

But to be able to do that, we would need to compute every single value of that column, lay it out (i.e. convert the value to an image), and then compute the max image width. This would take a long time, and it possibly make the entire UX sluggish.

That being said, there are probably smart ways to cheat 😄 . For instance:
 - we could limit that behaviors to tables under a certain length
 - we could give up past a certain width too: if we encounter an element that's, say 70 pixels wide, we can say "well, the column is going to be quite wide anyway, so let's just go for the biggest possible width"

@artaxxx what do you think?

**Comment by artaxxx** [comment]
Yes, this is definitely something we have to do! There are also column widths based on the longest word of any languages that should be adapted to the current language (Statistics > Stats : 1st column for instance).

Then, for editable tables we could imagine an editing field just as in Functions or Equations to avoid adapting dynamically the width when editing. And we could calculate the width with the value of the significant numbers in the settings. Today, we always display 7 figures but we should display more if the user set more significant numbers in the Settings.

**Comment by Stephan-P** [comment]
Reminder for v19

**Comment by Stephan-P** [comment]
Reminder for v20?

**Comment by Stephan-P** [comment]
Possibly the math notation i.e.1.234E56 could be of use here in writing large values in narrow columns
